File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKCS/PKCS06/ExtendedCertificateInfo.cs
using System;
using System.Runtime.Serialization;

//	ExtendedCertificateInfo ::= SEQUENCE {
//		version		INTEGER,
//		certificate Certificate,
//		attributes	Attributes 
//	}

namespace Aladdin.ASN1.ISO.PKCS.PKCS6
{
	[Serializable]
	public class ExtendedCertificateInfo : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<Integer			>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<PKIX.Certificate	>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<Attributes		    >().Factory(), Cast.N), 
		}; 
		// конструктор при сериализации
        protected ExtendedCertificateInfo(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public ExtendedCertificateInfo(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public ExtendedCertificateInfo(Integer version, PKIX.Certificate certificate, 
			Attributes attributes) : base(info, version, certificate, attributes) {}

		public Integer			Version		{ get { return (Integer			)this[0]; } } 
		public PKIX.Certificate	Certificate	{ get { return (PKIX.Certificate)this[1]; } }
		public Attributes		Attributes	{ get { return (Attributes		)this[2]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkcs/pkcs6/ExtendedCertificateInfo.java
package aladdin.asn1.iso.pkcs.pkcs6;
import aladdin.asn1.*; 
import aladdin.asn1.Integer; 
import aladdin.asn1.iso.*; 
import aladdin.asn1.iso.pkix.*; 
import java.io.*;

//	ExtendedCertificateInfo ::= SEQUENCE {
//		version		INTEGER,
//		certificate Certificate,
//		attributes	Attributes 
//	}

public final class ExtendedCertificateInfo extends Sequence<IEncodable>
{
    private static final long serialVersionUID = 5894748203188419219L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(Integer	.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(Certificate.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(Attributes	.class).factory(), Cast.N), 
	}; 
	// конструктор при раскодировании
	public ExtendedCertificateInfo(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public ExtendedCertificateInfo(Integer version, 
		Certificate certificate, Attributes attributes) 
	{
		super(info, version, certificate, attributes); 
	}
	public final Integer        version		() { return (Integer	)get(0); } 
	public final Certificate	certificate	() { return (Certificate)get(1); }
	public final Attributes     attributes	() { return (Attributes	)get(2); }
}
